### PR TITLE
Fix setting bezier track handle mode from inspector

### LINKED ISSUE #74224: Setting handle mode on bezier curve from the inspector does nothing
### Godot version

4.0.stable

### System information

Windows 10, Forward+, GTX 1070 driver 528.49

### Issue description

When trying to edit the handle mode of a bezier cruve key from the inspector tab, the mode doesn't change. It works fine when right clicking on the key and setting the handle mode from here though. 

https://user-images.githubusercontent.com/24484962/222379540-29d71b6d-ff0c-4a21-9905-9750d0581fda.mp4

### Steps to reproduce

On an animation track using a bezier curve, select a key and try to change the handle mode from the inspector tab

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by SaracenOne**
Closes https://github.com/godotengine/godot/issues/74224

### CODE REVIEW

File: editor/animation_track_editor.cpp [hunk 1]
```diff
@@ -306,8 +306,9 @@ bool AnimationTrackKeyEdit::_set(const StringName &p_name, const Variant &p_valu
 				setting = true;
 				undo_redo->create_action(TTR("Animation Change Keyframe Value"), UndoRedo::MERGE_ENDS);
 				int prev = animation->bezier_track_get_key_handle_mode(track, key);
-				undo_redo->add_do_method(this, "_bezier_track_set_key_handle_mode", animation.ptr(), track, key, value);
-				undo_redo->add_undo_method(this, "_bezier_track_set_key_handle_mode", animation.ptr(), track, key, prev);
+				ERR_FAIL_COND_V(!editor, false);
```
Comment (AThousandShips) [inline editor/animation_track_editor.cpp:309]:
```suggestion
				ERR_FAIL_NULL_V(editor, false);
```
Use proper error check

File: editor/animation_track_editor.cpp [hunk 2]
```diff
@@ -878,8 +879,10 @@ bool AnimationMultiTrackKeyEdit::_set(const StringName &p_name, const Variant &p
 							undo_redo->create_action(TTR("Animation Multi Change Keyframe Value"), UndoRedo::MERGE_ENDS);
 						}
 						int prev = animation->bezier_track_get_key_handle_mode(track, key);
-						undo_redo->add_do_method(this, "_bezier_track_set_key_handle_mode", animation.ptr(), track, key, value);
-						undo_redo->add_undo_method(this, "_bezier_track_set_key_handle_mode", animation.ptr(), track, key, prev);
+
+						ERR_FAIL_COND_V(!editor, false);
```
Comment (AThousandShips) [inline editor/animation_track_editor.cpp:883]:
```suggestion
						ERR_FAIL_NULL_V(editor, false);
```

**Comment by KoBeWi** [comment]
When you change the mode and it changes handles, doing undo will not undo handles.

**Comment by SaracenOne** [comment]
Should be fixed now!

File: editor/animation_track_editor.cpp [hunk 3]
```diff
@@ -306,10 +306,14 @@ bool AnimationTrackKeyEdit::_set(const StringName &p_name, const Variant &p_valu
...
-				undo_redo->add_do_method(this, "_bezier_track_set_key_handle_mode", animation.ptr(), track, key, value);
-				undo_redo->add_undo_method(this, "_bezier_track_set_key_handle_mode", animation.ptr(), track, key, prev);
+				int prev_mode = animation->bezier_track_get_key_handle_mode(track, key);
+				Vector2 prev_in_handle = animation->bezier_track_get_key_in_handle(track, key);
+				Vector2 prev_out_handle = animation->bezier_track_get_key_out_handle(track, key);
+				undo_redo->add_do_method(editor, "_bezier_track_set_key_handle_mode", animation.ptr(), track, key, value);
 				undo_redo->add_do_method(this, "_update_obj", animation);
+				undo_redo->add_undo_method(editor, "_bezier_track_set_key_handle_mode", animation.ptr(), track, key, prev_mode);
+				undo_redo->add_undo_method(animation.ptr(), "bezier_track_set_key_in_handle", track, key, prev_in_handle);
+				undo_redo->add_undo_method(animation.ptr(), "bezier_track_set_key_out_handle", track, key, prev_out_handle);
```
Comment (KoBeWi) [inline editor/animation_track_editor.cpp:316]: `_bezier_track_set_key_handle_mode` seems to have a check for null animation, which means it expects for some reason that the animation can be freed. This code does not have one. Can't say how safe is that.

**Review by KoBeWi — APPROVED** [review]

**Comment by YuriSizov** [comment]
Thanks!

**Comment by YuriSizov** [comment]
Cherry-picked for 4.2.2.

**Pull request merged**